include/VectorGraph.h

```diff
@@ -0,0 +1,739 @@
+	// returns true if the control window was clicked while in editing mode
+	bool isControlWindowPressed(int mouseY);
+	void processControlWindowPressed(int mouseX, int mouseY, bool isDragging, bool startMoving, int curX, int curY);
+	// returns -1 if no control / input was clicked
+	// returns displayed absolute control / input location based on inputCount
+	int getPressedControlInput(int mouseX, int mouseY, unsigned int controlCount);
+	// returns a float attrib value, valueOut = attrib value if it is a bool
+	float getInputAttribValue(unsigned int controlArrayLocation, bool* valueOut);
+	// sets the selected point's attrib to floatValue it it is float, else it sets the attrib to boolValue
+	void setInputAttribValue(unsigned int controlArrayLocation, float floatValue, bool boolValue);
+	// calculates the ideal text color
+	QColor getTextColorFromBaseColor(QColor baseColor);
+	// calculates a replacement background fill color
+	QColor getFillColorFromBaseColor(QColor baseColor);
+	// cuts the string to displayedLength(in px) size (estimated)
+	QString getTextFromDisplayLength(QString text, unsigned int displayLength);
+	// context menu actions
+	void addDefaultActions(QMenu* menu, QString controlDisplayText);
+
+	// inputDialog
+	std::pair<float, float> showCoordInputDialog();
+	float showInputDialog(float curInputValue);
+
+	// selection
+	// searches VectorGraphDataArray-s to select
+	// near clicked location
+	void selectData(int mouseX, int mouseY);
+	// searches for point in a given VectorGraphDataArray
+	// returns found location, when a point
+	// was found in the given distance
+	// else it returns -1
+	int searchForData(int mouseX, int mouseY, float maxDistance, VectorGraphDataArray* dataArray, bool isCurved);
+
+	// if the mouse is not moved
+	bool m_mousePress;
+	// decides addition or deletion
+	bool m_addition;
+
+	// radius, rx = ry
+	unsigned int m_pointSize;
+	unsigned int m_fontSize;
+	// draw simplified lines
+	bool m_isSimplified;
+	// true when applyDefaultColors is called, used when defaultColors are loading
+	bool m_isDefaultColorsApplyed;
+	QPixmap m_background;
+	// for 1 draw, it will use the VectorGraphDataArray
+	// m_bakedSamples without calling getSamples()
+	bool m_useGetLastSamples;
+
+	// if m_isLastSelectedArray == true then
+	// m_selectedArray can be used
+	// else if m_isSelected == false then
+	// m_selectedLocation amd m_selectedArray should not be used
+	unsigned int m_selectedLocation;
+	unsigned int m_selectedArray;
+	bool m_isSelected;
+	bool m_isCurveSelected;
+	// if m_selectedArray was the last array selected
+	bool m_isLastSelectedArray;
+
+	unsigned int m_graphHeight;
+	unsigned int m_controlHeight;
+	// displayed control count (+1 because of the ">>" button in editing mode)
+	unsigned int m_controlDisplayCount;
+	unsigned int m_controlDisplayPage;
+	bool m_isEditingActive;
+	std::vector<QString> m_controlText;
+	std::vector<QString> m_controlLineEffectText;
+	std::vector<bool> m_controlIsFloat;
```

Comment:
Will std::array be right? Or should I use c style arrays?

Reply:
If you know the size beforehand `std::array` will work. Do not use C-style arrays.
If we don't, then don't worry about changing this (yet).

Reply:
It is now std::array. Will tr() work?

Reply:
I don't know what you're asking about tr?

Reply:
For example `std::array<QString, 18> m_controlText` uses tr() for translation. Will memory be allocated properly when the translated text is larger?

Reply:
Why are you changing `m_controlText`? I was specifically talking about `std::vector<bool> m_controlIsFloat`, not about anything else.

Reply:
I am sorry I have accidentally misunderstood you. I think `m_controlText` should be changed from vector since it is never changed after construction. I just don't know how.